src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/InstanceNameTest/InstanceNameTest.cs

```diff
@@ -98,11 +98,23 @@ public static void PortNumberInSPNTestForTCP()
             Assert.True(port > 0, "Named instance must have a valid port number.");
             builder.DataSource = $"{builder.DataSource},{port}";
 
-            PortNumberInSPNTest(builder.ConnectionString, port);
+            PortNumberInSPNTest(connectionString: builder.ConnectionString, expectedPortNumber: port);
+        }
+
+        [ConditionalFact(nameof(IsSPNPortNumberTestForNP))]
+        public static void SPNTestForNPMustReturnNamedInstance()
+        {
+            string connectionString = DataTestUtility.NPConnectionString;
+            SqlConnectionStringBuilder builder = new(connectionString);
+
+            DataTestUtility.ParseDataSource(builder.DataSource, out _, out _, out string instanceName);
```

Comment:
Please unit test this particular method with wide variety of datasource names.
Include multiple test cases to validate parser with/without instancename/port/protocol etc.

Reply:
Unit tests for DataTestUtility.ParseDataSource were added for a total of 24 tests. Let me know if you want to add more tests.